pkg/cluster/cluster.go

```diff
@@ -610,18 +612,27 @@ func (c *Cluster) initSystemUsers() {
 func (c *Cluster) initRobotUsers() error {
 	for username, userFlags := range c.Spec.Users {
 		if !isValidUsername(username) {
-			return fmt.Errorf("invalid username: '%v'", username)
+			return fmt.Errorf("invalid username: %q", username)
 		}
 
 		flags, err := normalizeUserFlags(userFlags)
 		if err != nil {
-			return fmt.Errorf("invalid flags for user '%v': %v", username, err)
+			return fmt.Errorf("invalid flags for user %q: %v", username, err)
 		}
-
-		c.pgUsers[username] = spec.PgUser{
-			Name:     username,
-			Password: util.RandomPassword(constants.PasswordLength),
-			Flags:    flags,
+		if _, present := c.pgUsers[username]; !present {
+			c.pgUsers[username] = spec.PgUser{
+				Name:     username,
+				Password: util.RandomPassword(constants.PasswordLength),
+				Flags:    flags,
+			}
+		} else {
+			// avoid overwriting the password if the user is already there. The flags should be
+			// merged here, but since there is no mechanism to define them for non-robot roles
+			// they are assigned from the robot user.
+			c.logger.Debugf("merging user %q data", username)
```

Comment:
Shall this message also say in which places exactly user is defined twice ?

Reply:
good point.